File: pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <math.h>
#include <string.h>


int main(void){
    string s = get_string("Text: \n");
    int letters = 0, sentance = 0, words = 1;
    
    for (int i = 0, n = strlen(s); i < n; ++i){
        if (s[i] == ' '){
            words++;
        }
        else if (s[i] == '!' || s[i] == '?' || s[i] == '.'){
            sentance++;
        } else if (islower(s[i]) || isupper(s[i])){
            letters++;
        }
    }
    
    float answer = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentance / (float) words) - 15.8;
    if (answer < 16 && answer >= 0){
        printf("Grade %d\n", (int) round(answer));
    } else if (answer >= 16){
        printf("Grade 16+\n");
    } else {
        printf("Before Grade 1\n");
    }
    
}

File: pset6/readability.py
from cs50 import get_string

s = get_string("Text: ")
letters = 0
sentance = 0
words = 1

for i in range(len(s)):
    if s[i] == ' ' and s[i + 1]:
        words += 1
    elif s[i] == '.' or s[i] == '?' or s[i] == '.':
        sentance += 1
    elif s[i].isalpha():
        letters += 1


L = float(letters / words * 100)
S = float(sentance / words * 100)

# answer = round(0.0588 * (100 * float(letters) // float(words)) - 0.296 * (100 * float(sentance) // float(words)) - 15.8)
index = round((float)(0.0588 * L - 0.296 * S - 15.8))


if index < 16 and index >= 0:
    print(f"Grade {int(index)}")
elif index >= 16:
    print("Grade 16+")
else:
    print("Before Grade 1")
